rholang/src/rust/interpreter/pretty_printer.rs

```diff
@@ -1100,177 +1100,197 @@ impl PrettyPrinter {
 // rholang/src/test/scala/coop/rchain/rholang/interpreter/PrettyPrinterTest.scala
 #[cfg(test)]
 mod tests {
-    use crate::rust::interpreter::compiler::normalize::{normalize_match_proc, ProcVisitOutputs};
-    use crate::rust::interpreter::compiler::normalizer::ground_normalize_matcher::normalize_ground;
-    use crate::rust::interpreter::compiler::rholang_ast::{Collection, Eval, KeyValuePair, Proc};
-    use crate::rust::interpreter::errors::InterpreterError;
+    use crate::rust::interpreter::compiler::rholang_ast::{
+        ASTBuilder, AnnProc, KeyValuePair, Name, Proc, Var, Id, ProcRemainder
+    };
+    use crate::rust::interpreter::compiler::source_position::SourcePosition;
+    use crate::rust::interpreter::test_utils::utils::{
+        defaults, test, test_normalize_match_proc
+    };
     use crate::rust::interpreter::pretty_printer::PrettyPrinter;
-    use crate::rust::interpreter::test_utils::utils::collection_proc_visit_inputs_and_env;
-    use pretty_assertions;
+    use pretty_assertions::assert_eq;
+
+    // Custom test function for ground expressions
+    fn test_ground_expression(proc: &Proc, expected: &str) {
+        test_normalize_match_proc(
+            proc,
+            defaults(),
+            test("should pretty print correctly", |actual_par, free_map| {
+                let mut printer = PrettyPrinter::new();
+                let printed = printer.build_string_from_message(actual_par);
+                assert_eq!(printed, expected);
+                assert!(free_map.is_empty(), "Expected free map to be empty");
+            })
+        );
+    }
+
+    // Helper function to test collections with PrettyPrinter
+    fn test_pretty_print(name: &str, proc: &Proc, bound_shift: i32, expected: &str) {
+        test_normalize_match_proc(
+            proc,
+            defaults(),
+            test(name,|actual_par, _| {
+                let mut printer = PrettyPrinter::create(0, bound_shift);
+                let printed = printer.build_string_from_message(actual_par);
+                assert_eq!(printed, expected);
+            })
+        );
+    }
+
+    // Helper functions to reduce duplication
+    fn create_id(name: &str, pos: SourcePosition) -> Id {
```

Comment:
Well, I am not sure if this helper brings something to the table: `Id { name: "a", pos }` is not really longer from `create_id("a", pos)`. If anything, I feel this should be added to `impl Id`